Mapper/EntityMapper.php

```diff
@@ -299,6 +310,10 @@ function (CategoryStruct $category) use ($context, &$categorySort) {
             $releaseDate = $releaseDate->format(DateTimeInterface::ATOM);
         }
 
+        if (null !== ($creationDate = $product->getCreatedAt())) {
+            $creationDate = $creationDate->format(DateTimeInterface::ATOM);
+        }
```

Comment:
I faced a problem with the `releaseDate`. The value `null` (or an empty string) is not valid for Elasticsearch and leads to an error.

To avoid this add a valid default date:
```suggestion
        $creationDate = '0001-01-01T00:00:00';
        if (null !== ($creationDateObject = $product->getCreatedAt())) {
            $creationDate = $creationDateObject->format(DateTimeInterface::ATOM);
        }
```